src/core/public/base_path/base_path.mock.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { BasePathService, BasePathStart } from './base_path_service';
+
+const createStartContractMock = () => {
+  const startContract: jest.Mocked<BasePathStart> = {
+    get: jest.fn(),
+    addToPath: jest.fn(),
+    removeFromPath: jest.fn(),
+  };
+  startContract.get.mockReturnValue('get');
+  startContract.addToPath.mockReturnValue('addToPath');
+  startContract.removeFromPath.mockReturnValue('removeFromPath');
+  return startContract;
+};
+
+type MethodKeysOf<T> = {
```

Comment:
as this type definition is duplicated across all `*.mock.ts` files I'd suggest to finally decide where to can place such helpers. We can do that in following PRs tho

Reply:
I'm definitely pro adding a file of "meta" types that are types used to create other types. This shouldn't have any types that are Kibana or even Node/Browser specific, but just generic types like this.

Reply:
I don't want to introduce new entities in the codebase. how about to just place such helpers in `typings/index.d.ts` ?

Reply:
That seems like an appropriate place to me.